### ISSUE TITLE
Platform analyzer should honor <SupportedFramework> configuration

### ISSUE THREAD

**Issue by terrajobst** [dotnet/roslyn-analyzers#4017]
Note: This is currently blocked by https://github.com/dotnet/sdk/pull/12872.

We don't want to assume that a developer wants to run on all platforms all the time, which is why we [added the <SupportedFramework> MSBuild configuration](https://github.com/dotnet/designs/blob/master/accepted/2020/platform-exclusion/platform-exclusion.md#build-configuration-for-platforms). @mavasani made sure it's passed to the analyzer configuration, which we should query.

### Repro

```C#
static class Program
{
    public static void Main()
    {
        Some.ApiUnsupportedByWindows();
    }
}

class Some
{
    [UnsupportedOSPlatform("windows")]
    public static void ApiUnsupportedByWindows() { }
}
```

```XML
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <SupportedPlatform Remove="windows" />
  </ItemGroup>

</Project>
```

### Expected

No diagnostics

### Actual

> 'ApiUnsupportedByWindows' is not supported or has been removed from windows

**Comment by buyaa-n** [comment]
Yes, it is not added to the analyzer yet, i thought it is not needed for 5.0, is this need for 5.0 @terrajobst ?

**Comment by terrajobst** [comment]
> Yes, it is not added to the analyzer yet, i thought it is not needed for 5.0, is this need for 5.0 @terrajobst ?

Yes, it's key for the Blazor/Browser scenario.

**Comment by buyaa-n** [comment]
Fixed and merged to RC1, closing
